lib/object.c

```diff
@@ -196,21 +196,36 @@ static tool_rc tpm2_util_object_fetch_parent_from_tpk(const char *objectstr,
     tool_rc rc = tool_rc_general_error;
     BIO *input_bio = 0;
     TSSPRIVKEY_OBJ *tpk = 0;
+    BIGNUM *bn_parent = 0;
```

Comment:
NULL, @idesai knock off this 0 pointer non-sense :-p. stdlib.h defines the proper macro, "NULL" for "zeroing" of a pointer.

Reply:
https://www.stroustrup.com/bs_faq2.html#null

Reply:
> https://www.stroustrup.com/bs_faq2.html#null

@malikabhi05 you can use NULL, no problem :)

Reply:
> https://www.stroustrup.com/bs_faq2.html#null

Barne Stroustrup is the creator of C++ and his blog is for C++, we are C programmers. Therefore we follow the K&R Book on C: https://www.amazon.com/Programming-Language-2nd-Brian-Kernighan/dp/0131103628, which states on page 102:

"Pointers and Integers are not interchangeable. Zero is the sole exception: the constant zero may be assigned to a pointer, and a pointer may be compared with the constant zero. The symbolic constant NULL is often used in the place of zero , as a mnemonic to indicate more clearly that this is a special value for a pointer."

Now in Bourne's blog he mentions, "In pre-standard code, NULL was/is sometimes defined to something unsuitable and therefore had/has to be avoided. That's less common these **days**". I have been bit by that before where 0 was a valid and in use address (which made things like `if(!fooptr) { /* do this /* }` to be broken, so its very not wise to do that as porting software is a nightmare.